linuxKernel.packages.linux_zen.ddcci-driver 6.4.1 fails to build.

### Describe the bug

linuxKernel.packages.linux_zen 6.4.1 fails to build with ddcci-driver, presumably because of  https://gitlab.com/ddcci-driver-linux/ddcci-driver-linux/-/issues/34.

```
/build/source/ddcci/ddcci.c:43:34: error: macro "DEFINE_SEMAPHORE" requires 2 arguments, but only 1 given
   43 | static DEFINE_SEMAPHORE(core_lock);
```

### Steps To Reproduce

Vuild linuxKernel.packages.linux_zen.ddcci-driver

### Notify maintainers

@erdnaxe 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.3.1-zen1, NixOS, 23.05 (Stoat), 23.05.1748.e11142026e2`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos-23.05, nixos-unstable"`
 - channels(mabl): `"home-manager-23.05.tar.gz, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/os-specific/linux/ddcci/default.nix
+++ b/pkgs/os-specific/linux/ddcci/default.nix
@@ -1,25 +1,16 @@
-{ lib, stdenv, fetchFromGitLab, kernel, fetchpatch }:
+{ lib, stdenv, fetchFromGitLab, kernel }:
 
 stdenv.mkDerivation rec {
   pname = "ddcci-driver";
-  version = "0.4.3";
+  version = "0.4.4";
   name = "${pname}-${kernel.version}-${version}";
 
   src = fetchFromGitLab {
     owner = "${pname}-linux";
     repo = "${pname}-linux";
     rev = "v${version}";
-    hash = "sha256-1Z6V/AorD4aslLKaaCZpmkD2OiQnmpu3iroOPlNPtLE=";
+    hash = "sha256-4pCfXJcteWwU6cK8OOSph4XlhKTk289QqLxsSWY7cac=";
   };
-
-  patches = [
-    # https://gitlab.com/ddcci-driver-linux/ddcci-driver-linux/-/merge_requests/12
-    (fetchpatch {
-      name = "kernel-6.2-6.3.patch";
-      url = "https://gitlab.com/ddcci-driver-linux/ddcci-driver-linux/-/commit/1ef6079679acc455f75057dd7097b5b494a241dc.patch";
-      hash = "sha256-2C2leS20egGY3J2tq96gsUQXYw13wBJ3ZWrdIXxmEYs=";
-    })
-  ];
 
   hardeningDisable = [ "pic" ];
 

